woudc_data_registry/epicentre/instrument.py

```diff
@@ -188,13 +187,26 @@ def update(ctx, identifier, station, dataset,
     """Update instrument information"""
 
     instrument_ = {}
+    registry_ = registry.Registry()
 
     if station:
         instrument_['station_id'] = station
     if dataset:
         instrument_['dataset_id'] = dataset
-    if station and contributor:
-        instrument_['deployment_id'] = ':'.join([station, contributor])
+    if station or contributor:
+        temp_index = registry_.query_full_index(Instrument)
+        i = 0
+        for i in range(len(temp_index)):
+            if temp_index[i].instrument_id == identifier:
+                break
+        if station and (not contributor):
+            cbtr = temp_index[i].deployment.contributor_id
+            instrument_['deployment_id'] = ':'.join([station, cbtr])
+        elif contributor and (not station):
+            stn = temp_index[i].station_id
+            instrument_['deployment_id'] = ':'.join([stn, contributor])
+        elif contributor and station:
+            instrument_['deployment_id'] = ':'.join([station, contributor])
```

Comment:
You can re-structure the if statements so that this case (both contributor and station are provided) doesn't cause that big, expensive search in the registry.